### DISCUSSION TITLE
Allow positioning the symbol on the right of the amount

### DISCUSSION THREAD

**Discussion by pierrexp9** [dm4t2/vue-currency-input#332] [Ideas]
Hello,
European guy here.
In many countries, we actually place the € symbol on the right, after a space.

It would be great if we could configure the position and the space for the symbol :)

Thanks for the great work guys !

Reference : 
https://publications.europa.eu/code/en/en-370303.htm
excerpt : 
**Position of the euro sign (€) in amounts**
In all other official EU languages the order is reversed; the amount is followed by a hard space and the euro sign:
_une somme de 30 €_

**Comment by dm4t2** [comment]
The position of the currency symbol is automatically managed by the ECMA Internationalization API and only depends on the configured `locale`, for example `"fr"`. 

This is a key feature of Vue Currency Input so that you don't have to care about the "right" locale dependent formatting.

**Reply by pierrexp9** [reply]
Hi dm4t2, and thank you for getting back to me so quickly !

Actually, the ECMA Internationalization API seems to not be aware of the usual way of positioning the symbol in most of the European countries.
It only works for Dutch, Irish and Maltese, but for all the other European countries, as you can see in the reference I provided, the positioning of the symbol is different.

In dutch you write  :  **€30**
And here, the Vue Currency and the ECMA I18 API has it right !

But in France, Germany, Italy, Spain, Belgium, Finland, Portugal, etc.. basically all the other countries, you write it that way : 
**30 €**

You may also read this paragraph on wikipedia, even if I think the first reference I provided you with is the most official you will get.

https://en.wikipedia.org/wiki/Euro_sign
_Placement of the sign varies. Countries have generally continued the style used for their former currencies. In those countries where previous convention was to place the currency sign before the figure, the euro sign is placed in the same position (e.g., €3.50). **In those countries where the amount preceded the national currency sign, the euro sign is again placed in that relative position (e.g., 3,50 €)**._

The second sentence basically talks about 90% of the countries here in Europe

**Reply by dm4t2** [reply]
Your described behavior is already covered by `Intl.NumberFormat` and works as expected in the Chrome console:

<img width="391" alt="image" src="https://user-images.githubusercontent.com/11592259/174498491-a0ff1cda-ffd6-455b-a787-ae6939a218c3.png">

**Reply by pierrexp9** [reply]
ok my bad, I assumed I was already using the "fr" locale ( I thought VueCurrency was using the lang attribute on the html element, but I read it's using the browser language, and mine is set to english)

Now that I force it manually in the config, it works perfectly!

Thanks a lot for your follow up !

May I suggest that VueCurrency should default to the html lang attribute instead of the browser language? Or do you think it would be wrong to do it this way ?

**Reply by dm4t2** [reply]
> May I suggest that VueCurrency should default to the html lang attribute instead of the browser language? Or do you think it would be wrong to do it this way ?

This be should be implemented by your component/app: 
```
useCurrencyInput({ locale: document.querySelector('html').getAttribute('lang') })
```

**Comment by pierrexp9** [comment]
Thanks, I indeed implemented it this way. I'll take that as a no for my suggestion :) 
My thinking is that it was more logic to use the lang attribute than the browser language, just wanted to share.

Anyway, thanks a lot for taking the time for me! And thanks a lot for the library!
